internal/http.go

```diff
@@ -0,0 +1,206 @@
+package internal
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"os/signal"
+	"path"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/jackc/pgx/v4"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+const (
+	postgresURL    = "host=postgresql.test-project.svc  port=5432 user=curator password=M5rBgWkN8LfjeyI8 dbname=curatordb sslmode=disable"
+	httpServerPort = ":5000"
+)
+
+var (
+	reportLog = ctrl.Log.WithName("report")
+)
```

Comment:
nit: Let's remove having a logger live as a global variable - any functions that need access to a logger should accept it as a parameter imo.